packages/mwp-language-plugin/src/util/getLanguage.js

```diff
@@ -29,7 +29,16 @@ export const getCookieLang: ParseRequestLang = (request: HapiRequest) => {
 export const getUrlLang: ParseRequestLang = (request: HapiRequest) => {
 	const { supportedLangs } = getServerSettings(request);
 	const urlLang = request.url.path.split('/')[1];
-	return supportedLangs.includes(urlLang) && urlLang;
+	const validRequestLang = supportedLangs.includes(urlLang) && urlLang;
+
+	// If request url language is invalid, look at referer for language code
+	// This is for cases when the request url is the api proxy path ie `/mu_api/`
+	if (!validRequestLang && request.headers['referer']) {
+		const refererUrl = new URL(request.headers['referer']);
+		const refererLang = refererUrl.pathname.split('/')[1];
+		return supportedLangs.includes(refererLang) && refererLang;
+	}
+	return validRequestLang;
```

Comment:
General question, does the platform code then set a language cookie after it parses the language from a url? I'm going to assume yes, but if not, we should.

Reply:
it does not- currently all the cookie setting happens from chapstick. i'd actually love to have a discussion soon generally about the language cookie/param/pref landscape and try to come up with a better overall approach

Reply:
I think that was the underlying cause of this bug - when the code was first written, I don't think there were cases where you could get to mup-web without having a MEETUP_LANGUAGE cookie, but now there are. Definitely worth a discussion to define what the expected behavior should be in different cases